java-manta-client/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -1478,61 +1479,127 @@ public void putSnapLink(final String rawLinkPath, final String rawObjectPath,
     }
 
     /**
-     * Moves a file from one path to another path. This operation is not
-     * transactional and will fail or produce inconsistent result if the source
-     * or the destination is modified while the operation is in progress.
+     * <p>Moves an object from one path to another path. When moving
+     * directories or files between different directories, this operation is
+     * not transactional and may fail or produce inconsistent result if
+     * the source or the destination is modified while the operation is
+     * in progress.</p>
+     *
+     * <p><em>Note: If you need to do a rename, then use
+     * {@link #putSnapLink(String, String, MantaHttpHeaders)}</em></p>
      *
      * @param source Original path to move from
      * @param destination Destination path to move to
      * @throws IOException thrown when something goes wrong
      */
     public void move(final String source, final String destination)
             throws IOException {
+        move(source, destination, false);
+    }
+
+    /**
+     * <p>Moves an object from one path to another path. When moving
+     * directories or files between different directories, this operation is
+     * not transactional and may fail or produce inconsistent result if
+     * the source or the destination is modified while the operation is
+     * in progress.</p>
+     *
+     * <p><em>Note: If you need to do a rename, then use
+     * {@link #putSnapLink(String, String, MantaHttpHeaders)}</em></p>
+     *
+     * @param source Original path to move from
+     * @param destination Destination path to move to
+     * @param recursivelyCreateDestinationDirectories when true create the full destination directory path
+     * @throws IOException thrown when something goes wrong
+     */
+    public void move(final String source, final String destination,
```

Comment:
Is there a reason these parameters shouldn't be Validated? I think that is a pre-existing omission and I'm fine with a separate change.